drivers/sensor/wsen_tids_2521020222501/wsen_tids_2521020222501_trigger.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_tids_2521020222501
+
+#include <stdlib.h>
+
+#include <zephyr/logging/log.h>
+
+#include "wsen_tids_2521020222501.h"
+
+LOG_MODULE_DECLARE(WSEN_TIDS_2521020222501, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Enable/disable interrupt handling */
+static inline void tids_2521020222501_setup_interrupt(const struct device *dev, bool enable)
+{
+	const struct tids_2521020222501_config *cfg = dev->config;
+	unsigned int flags = enable ? GPIO_INT_EDGE_TO_ACTIVE : GPIO_INT_DISABLE;
+
+	gpio_pin_interrupt_configure_dt(&cfg->interrupt_gpio, flags);
+}
+
+/*
+ * Is called when an interrupt has occurred.
+ */
+static void tids_2521020222501_handle_interrupt(const struct device *dev)
+{
+	struct tids_2521020222501_data *data = dev->data;
+
+	/* Disable interrupt handling until the interrupt has been processed */
+	tids_2521020222501_setup_interrupt(dev, false);
+
+#if defined(CONFIG_WSEN_TIDS_2521020222501_TRIGGER_OWN_THREAD)
+	k_sem_give(&data->sem);
+#elif defined(CONFIG_WSEN_TIDS_2521020222501_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&data->work);
+#endif
+}
+
+/*
+ * Calls trigger handles.
+ */
+static void tids_2521020222501_process_interrupt(const struct device *dev)
+{
+	struct tids_2521020222501_data *data = dev->data;
+	struct sensor_trigger temperature_low_trigger = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_TIDS_2521020222501_THRESHOLD_LOWER};
+	struct sensor_trigger temperature_high_trigger = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_TIDS_2521020222501_THRESHOLD_UPPER};
```

Comment:
Need to save a pointer to the user's sensor trigger and pass it back, rather than create a new sensor trigger on the stack. See #55130

Reply:
The trigger pointer is now stored